linera-sdk/src/contract/conversions_to_wit.rs

```diff
@@ -184,3 +188,80 @@ impl From<log::Level> for wit_system_api::LogLevel {
         }
     }
 }
+
+impl From<PublicKey> for wit_system_api::PublicKey {
+    fn from(public_key: PublicKey) -> Self {
+        let parts = <[u64; 4]>::from(public_key);
+
+        Self {
+            part1: parts[0],
+            part2: parts[1],
+            part3: parts[2],
+            part4: parts[3],
+        }
+    }
+}
+
+impl From<TimeDelta> for wit_system_api::TimeDelta {
+    fn from(delta: TimeDelta) -> Self {
+        Self {
+            inner0: delta.as_micros(),
+        }
+    }
+}
+
+impl From<TimeoutConfig> for wit_system_api::TimeoutConfig {
+    fn from(config: TimeoutConfig) -> Self {
+        let TimeoutConfig {
+            fast_round_duration,
+            base_timeout,
+            timeout_increment,
+            fallback_duration,
+        } = config;
+        Self {
+            fast_round_duration: fast_round_duration.map(Into::into),
+            base_timeout: base_timeout.into(),
+            timeout_increment: timeout_increment.into(),
+            fallback_duration: fallback_duration.into(),
+        }
+    }
+}
+
+impl From<ApplicationPermissions> for wit_system_api::ApplicationPermissions {
+    fn from(permissions: ApplicationPermissions) -> Self {
+        let ApplicationPermissions {
+            execute_operations,
+            mandatory_applications,
+            close_chain,
+        } = permissions;
+        Self {
+            execute_operations: execute_operations
+                .map(|ea| ea.into_iter().map(Into::into).collect()),
```

Comment:
Optional nit: maybe a slightly clearer name?

```suggestion
                .map(|ops| ops.into_iter().map(Into::into).collect()),
```

Reply:
(Or maybe `app_ids`: It's the applications whose operations are allowed.)